modules/gfac/gfac-core/src/main/java/org/apache/airavata/gfac/core/context/ProcessContext.java

```diff
@@ -681,7 +687,7 @@ public String getQualityOfService() {
 	public String getQueueName() {
 		if (isUseUserCRPref() &&
 				userComputeResourcePreference != null &&
-				isValid(userComputeResourcePreference.getPreferredBatchQueue())) {
+				userComputeResourcePreference.getPreferredBatchQueue() != null) {
```

Comment:
Why switch from `isValid` to `!= null`? Is it because the empty string is valid here (I wouldn't think it is)?  Same question on lines 658 and 679.

Reply:
if the isUseUserCRPref() is true, we should fetch whatever we have in userComputeResourcePreference. Issue for Airavata 2143 is being some values as "" in userComputeResourcePreference and fetching them from gateway preferences. Checking for null is just a sanity check.